cedar-policy-cli/src/lib.rs

```diff
@@ -1010,14 +1095,42 @@ fn read_json_policy(filename: Option<impl AsRef<Path> + std::marker::Copy>) -> R
         .wrap_err_with(|| format!("failed to create policy set from {context}"))
 }
 
-fn read_schema_file(filename: impl AsRef<Path> + std::marker::Copy) -> Result<Schema> {
+fn read_schema_file(
+    filename: impl AsRef<Path> + std::marker::Copy,
+    format: SchemaFormat,
+) -> Result<Schema> {
     let schema_src = read_from_file(filename, "schema")?;
-    Schema::from_str(&schema_src).wrap_err_with(|| {
-        format!(
-            "failed to parse schema from file {}",
-            filename.as_ref().display()
-        )
-    })
+    match format {
+        SchemaFormat::Json => Schema::from_str(&schema_src).wrap_err_with(|| {
+            format!(
+                "failed to parse schema from file {}",
+                filename.as_ref().display()
+            )
+        }),
+        SchemaFormat::Human => {
+            let custom_schema =
+                cedar_policy_validator::custom_schema::parser::parse_schema(&schema_src)
+                    .map_err(|err| {
+                        Report::new(err).with_source_code(NamedSource::new(
+                            filename.as_ref().to_string_lossy(),
+                            schema_src,
+                        ))
+                    })
+                    .wrap_err_with(|| {
+                        format!(
+                            "failed to parse schema from file {}",
+                            filename.as_ref().display()
+                        )
+                    })?;
+            let (json_schema, _) = custom_schema_to_json_schema(custom_schema)
+                .map_err(|err| miette!("failed to translate custom schema: {err:?}"))?;
+            Schema::from_str(
+                &serde_json::to_string(&json_schema)
+                    .map_err(|err| miette!("failed to serialize schema: {err}"))?,
+            )
```

Comment:
why is CLI code converting from `custom_schema` to `json_schema` and then calling `Schema::from_str()`?  The fact that custom schemas are parsed by first converting to JSON seems like it should be an implementation detail of the `validator` crate, and not something that external users (like CLI) should need to know or be aware of